### ISSUE TITLE
Error when running read corection

### ISSUE THREAD

**Issue by dmacguigan** [nanoporetech/dorado#864]
# Issue Report

## Please describe the issue:

Hello,

I am attempting to run HERRO read correction with Dorado 0.7.0 on a computing cluster with A100 and H100 compute nodes. However, I encountered the following error.

```
terminate called after throwing an instance of 'c10::DynamicLibraryError'
```

Do you have any advice on how to proceed?

## Steps to reproduce the issue:

*Please list any steps to reproduce the issue.*

## Run environment:
- Dorado version: 0.7.0
- Dorado command: 
```
# INPUTS ################################################
WD="/vscratch/grp-tkrabben/MacGuigan/rawNanoporeData/Svit_TJK-76/Dorado" # working directory
READS="Svit_TJK76.cell1.passed.fastq.gz" # uncorrected reads, must be located in WD
MIN_READ_LEN=10000 # min read length (bp), HERRO documentation suggests >10 kb
SP="Svit_TJK76" # species or sample prefix
CORR_READS="Svit_TJK76.cell1.L10000.HERRO.fasta" # name for corrected reads, must be .fasta
THREADS=56

# filter by min read length
fastp -i ${READS} -o ${SP}_L${MIN_READ_LEN}.fastq.gz -h ${SP}_L${MIN_READ_LEN}.html -j ${SP}_L${MIN_READ_LEN}.json -l ${MIN_READ_LEN} --disable_quality_filtering --disable_adapter_trimming --disable_trim_poly_g -V -w 8

# convert to bgzip
gunzip ${SP}_L${MIN_READ_LEN}.fastq.gz
bgzip ${SP}_L${MIN_READ_LEN}.fastq

# run read correction
/projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/dorado correct -t ${THREADS} ${SP}_L${MIN_READ_LEN}.fastq.gz > ${CORR_READS}
``` 
- Operating system: Ubuntu 22.04.4 LTS
- Hardware (CPUs, Memory, GPUs): 56 cores, 400 Gb RAM, A100 or H100 GPU
- Source data type (e.g., pod5 or fast5 - please note we always recommend converting to pod5 for optimal basecalling performance): FASTQ
- Source data location (on device or networked drive - NFS, etc.): on device
- Details about data (flow cell, kit, read lengths, number of reads, total dataset size in MB/GB/TB): 27.8 Gbp, R10.4.1 cell, Q20 chemistry
- Dataset to reproduce, if applicable (small subset of data to share as a pod5 to reproduce the issue):

## Logs

```
[2024-06-01 14:33:43.714] [info] Running: "correct" "-t" "56" "Svit_TJK76_L10000.fastq.gz"
[2024-06-01 14:33:43.732] [warning] Unknown certs location for current distribution. If you hit download issues, use the envvar `SSL_CERT_FILE` to specify the location manually.
[2024-06-01 14:33:43.734] [info]  - downloading herro-v1 with httplib
[2024-06-01 14:50:53.121] [info] > starting correction
terminate called after throwing an instance of 'c10::DynamicLibraryError'
  what():  Error in dlopen for library libnvrtc.so.11.2and libnvrtc-672ee683.so.11.2
Exception raised from DynamicLibrary at /pytorch/pyold/aten/src/ATen/DynamicLibrary.cpp:35 (most recent call first):
frame #0: c10::Error::Error(c10::SourceLocation, std::string) + 0x57 (0x1518fefac9b7 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #1: <unknown function> + 0x39f139f (0x1518f7f6539f in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #2: <unknown function> + 0x89889e2 (0x1518fcefc9e2 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #3: <unknown function> + 0x8988e32 (0x1518fcefce32 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #4: torch::jit::fuser::cuda::codegenOutputQuery(cudaDeviceProp const*, int&, int&, bool&) + 0x37 (0x1518feef5f97 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #5: torch::jit::tensorexpr::CudaCodeGen::CompileToNVRTC(std::string const&, std::string const&) + 0x5e (0x1518fef053de in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #6: torch::jit::tensorexpr::CudaCodeGen::Initialize() + 0x1f57 (0x1518fef0cbf7 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #7: <unknown function> + 0xa9a5268 (0x1518fef19268 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #8: torch::jit::tensorexpr::CreateCodeGen(std::string const&, std::shared_ptr<torch::jit::tensorexpr::Stmt>, std::vector<torch::jit::tensorexpr::CodeGen::BufferArg, std::allocator<torch::jit::tensorexpr::CodeGen::BufferArg> > const&, c10::Device, std::string const&) + 0x9b (0x1518fc19b5ab in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #9: torch::jit::tensorexpr::TensorExprKernel::compile() + 0x1ec7 (0x1518fc2c0217 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #10: torch::jit::tensorexpr::TensorExprKernel::TensorExprKernel(std::shared_ptr<torch::jit::Graph> const&, std::string const&, std::unordered_map<c10::Symbol, std::function<torch::jit::tensorexpr::Tensor (std::vector<c10::variant<torch::jit::tensorexpr::BufHandle, torch::jit::tensorexpr::VarHandle, double, long, bool, std::vector<torch::jit::tensorexpr::BufHandle, std::allocator<torch::jit::tensorexpr::BufHandle> >, std::vector<double, std::allocator<double> >, std::vector<long, std::allocator<long> >, std::string, c10::monostate>, std::allocator<c10::variant<torch::jit::tensorexpr::BufHandle, torch::jit::tensorexpr::VarHandle, double, long, bool, std::vector<torch::jit::tensorexpr::BufHandle, std::allocator<torch::jit::tensorexpr::BufHandle> >, std::vector<double, std::allocator<double> >, std::vector<long, std::allocator<long> >, std::string, c10::monostate> > > const&, std::vector<torch::jit::tensorexpr::ExprHandle, std::allocator<torch::jit::tensorexpr::ExprHandle> > const&, std::vector<torch::jit::tensorexpr::ExprHandle, std::allocator<torch::jit::tensorexpr::ExprHandle> > const&, c10::optional<c10::ScalarType> const&, c10::Device)>, std::hash<c10::Symbol>, std::equal_to<c10::Symbol>, std::allocator<std::pair<c10::Symbol const, std::function<torch::jit::tensorexpr::Tensor (std::vector<c10::variant<torch::jit::tensorexpr::BufHandle, torch::jit::tensorexpr::VarHandle, double, long, bool, std::vector<torch::jit::tensorexpr::BufHandle, std::allocator<torch::jit::tensorexpr::BufHandle> >, std::vector<double, std::allocator<double> >, std::vector<long, std::allocator<long> >, std::string, c10::monostate>, std::allocator<c10::variant<torch::jit::tensorexpr::BufHandle, torch::jit::tensorexpr::VarHandle, double, long, bool, std::vector<torch::jit::tensorexpr::BufHandle, std::allocator<torch::jit::tensorexpr::BufHandle> >, std::vector<double, std::allocator<double> >, std::vector<long, std::allocator<long> >, std::string, c10::monostate> > > const&, std::vector<torch::jit::tensorexpr::ExprHandle, std::allocator<torch::jit::tensorexpr::ExprHandle> > const&, std::vector<torch::jit::tensorexpr::ExprHandle, std::allocator<torch::jit::tensorexpr::ExprHandle> > const&, c10::optional<c10::ScalarType> const&, c10::Device)> > > >, std::vector<long, std::allocator<long> >, bool, std::unordered_map<torch::jit::Value const*, std::vector<torch::jit::StrideInput, std::allocator<torch::jit::StrideInput> >, std::hash<torch::jit::Value const*>, std::equal_to<torch::jit::Value const*>, std::allocator<std::pair<torch::jit::Value const* const, std::vector<torch::jit::StrideInput, std::allocator<torch::jit::StrideInput> > > > >) + 0x708 (0x1518fc2c0bc8 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #11: <unknown function> + 0x7a27d78 (0x1518fbf9bd78 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #12: <unknown function> + 0x7a238bc (0x1518fbf978bc in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #13: <unknown function> + 0x7a89263 (0x1518fbffd263 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #14: <unknown function> + 0x7a893b1 (0x1518fbffd3b1 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #15: <unknown function> + 0x7a7cd1c (0x1518fbff0d1c in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #16: <unknown function> + 0x7a86c53 (0x1518fbffac53 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #17: <unknown function> + 0x7a876fd (0x1518fbffb6fd in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #18: <unknown function> + 0x7a8794f (0x1518fbffb94f in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #19: <unknown function> + 0x7a87a20 (0x1518fbffba20 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #20: <unknown function> + 0x7a87274 (0x1518fbffb274 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #21: <unknown function> + 0x7a876fd (0x1518fbffb6fd in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #22: <unknown function> + 0x7a8794f (0x1518fbffb94f in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #23: <unknown function> + 0x7a88248 (0x1518fbffc248 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #24: torch::jit::Code::Code(std::shared_ptr<torch::jit::Graph> const&, std::string, unsigned long) + 0x52 (0x1518fbfee4c2 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #25: <unknown function> + 0x7ab1781 (0x1518fc025781 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #26: torch::jit::ProfilingGraphExecutorImpl::getOptimizedPlanFor(std::vector<c10::IValue, std::allocator<c10::IValue> >&, c10::optional<unsigned long>) + 0xa81 (0x1518fc024f81 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #27: torch::jit::ProfilingGraphExecutorImpl::getPlanFor(std::vector<c10::IValue, std::allocator<c10::IValue> >&, c10::optional<unsigned long>) + 0x79 (0x1518fc025529 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #28: <unknown function> + 0x7a6ba8a (0x1518fbfdfa8a in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #29: torch::jit::Method::operator()(std::vector<c10::IValue, std::allocator<c10::IValue> >, std::unordered_map<std::string, c10::IValue, std::hash<std::string>, std::equal_to<std::string>, std::allocator<std::pair<std::string const, c10::IValue> > > const&) const + 0x14e (0x1518fbc1c58e in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #30: /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/dorado() [0x8a9ba1]
frame #31: /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/dorado() [0x89e7ed]
frame #32: /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/dorado() [0x8a01d0]
frame #33: <unknown function> + 0x1196e380 (0x151905ee2380 in /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/../lib/libdorado_torch_lib.so)
frame #34: <unknown function> + 0x7de0 (0x1518f410fde0 in /lib64/libpthread.so.0)
frame #35: clone + 0x3f (0x1518f34fa54f in /lib64/libc.so.6)

/var/spool/slurmd/job15953227/slurm_script: line 66: 603372 Aborted                 (core dumped) /projects/academic/tkrabben/software/Dorado/dorado-0.7.0-linux-x64/bin/dorado correct -t ${THREADS} ${SP}_L${MIN_READ_LEN}.fastq.gz > ${CORR_READS}
```

**Comment by opplatek** [comment]
+1 

Same issue here. I used the precompiled Dorado v0.7.0 libraries. Ubuntu 18.04.6. NVIDIA GeForce RTX 2080 Ti. NVIDIA-SMI 530.41.03.

Dorado v0.5.0 basecalling on the same device works fine. I haven't tried basecalling with Dorado v0.7.0.

**Comment by tijyojwad** [comment]
duplicate of https://github.com/nanoporetech/dorado/issues/844 . a patch release of this will go out today that fixes the issue.

**Comment by tijyojwad** [comment]
Hi - fix has been deployed with dorado 0.7.1 - https://github.com/nanoporetech/dorado?tab=readme-ov-file#installation
